nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/s3/AbstractS3Processor.java

```diff
@@ -198,6 +218,88 @@ protected AmazonS3Client createClient(final ProcessContext context, final AWSCre
+        }
+
+        try {
+            return Region.getRegion(Regions.fromName(regionValue));
+        } catch (Exception e) {
+            throw new ProcessException(format("The [%s] attribute contains an invalid region value [%s]", S3_REGION_ATTRIBUTE, regionValue), e);
+        }
+    }
+
+    protected void setClientBasedOnRegionAttribute(final ProcessContext context, final Map<String, String> attributes) {
+        String regionValue = context.getProperty(S3_CUSTOM_REGION).getValue();
+
+        if (ATTRIBUTE_DRIVEN_REGION.getValue().equals(regionValue)) {
+            regionValue = attributes.get(S3_REGION_ATTRIBUTE);
+            Region region = parseRegionValue(regionValue);
+
+            final AWSConfiguration awsConfiguration = getConfiguration(context, region);
+            this.client = awsConfiguration.getClient();
+            this.region = awsConfiguration.getRegion();
+        }
+    }
+
+    protected AWSConfiguration getConfiguration(final ProcessContext context, final Map<String, String> attributes) {
+        final Region region = resolveRegion(context, attributes);
+        return super.getConfiguration(context, region);
+    }
+
+    @Override
+    public List<ConfigVerificationResult> verify(final ProcessContext context, final ComponentLog verificationLogger, final Map<String, String> attributes) {
+        final List<ConfigVerificationResult> results = new ArrayList<>();
+
+        try {
+            getConfiguration(context, attributes);
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(Outcome.SUCCESSFUL)
+                    .verificationStepName("Create S3 Client")
+                    .explanation("Successfully created S3 Client")
+                    .build());
+        } catch (final Exception e) {
+            verificationLogger.error("Failed to create S3 Client", e);
+            results.add(new ConfigVerificationResult.Builder()
+                    .outcome(Outcome.FAILED)
+                    .verificationStepName("Create S3 Client")
+                    .explanation("Failed to crete S3 Client: " + e.getMessage())
+                    .build());
+        }
+
+        return results;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        if (!isAttributeDrivenRegion(context)) {
+            setClientAndRegion(context);
+        }
+    }
+
+    private boolean isAttributeDrivenRegion(final ProcessContext context) {
+        String regionValue = context.getProperty(S3_CUSTOM_REGION).getValue();
+        return ATTRIBUTE_DRIVEN_REGION.getValue().equals(regionValue);
+    }
+
+    private static AllowableValue[] getCustomAvailableRegions() {
+        final List<AllowableValue> values = new ArrayList<>();
+        Collections.addAll(values, AbstractAWSCredentialsProviderProcessor.getAvailableRegions());
+        values.add(ATTRIBUTE_DRIVEN_REGION);
+        return values.toArray(new AllowableValue[0]);
+    }
```

Comment:
```suggestion
    private static AllowableValue[] getCustomAvailableRegions() {
    List<AllowableValue> values = new ArrayList<>(Arrays.asList(AbstractAWSCredentialsProviderProcessor.getAvailableRegions()));
    values.add(ATTRIBUTE_DRIVEN_REGION);
    return values.toArray(new AllowableValue[0]);
}
```

Reply:
We can also use Apache Commons ArrayUtils::addAll

```suggestion
    private static AllowableValue[] getCustomAvailableRegions() {
    AllowableValue[] availableRegions = AbstractAWSCredentialsProviderProcessor.getAvailableRegions();
    AllowableValue[] customRegions = new AllowableValue[]{ATTRIBUTE_DRIVEN_REGION};
    return ArrayUtils.addAll(availableRegions, customRegions);
}
```